From: Thomas Mingarelli <Thomas.Mingarelli@hp.com>
Date: Wed, 17 Mar 2010 15:33:31 +0000 (+0000)
Subject: hpwdt - fix lower timeout limit
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fwim%2Flinux-2.6-watchdog.git;a=commitdiff_plain;h=bd68f68c8a1e7129ea6f6dcb18f236529a927ff8
Patch-mainline: not yet
References: bnc#581103

[WATCHDOG] hpwdt - fix lower timeout limit

[Novell Bug 581103] HP Watchdog driver has arbitrary (wrong) timeout limits.
Fix the lower timeout limit to a more appropriate value.

Signed-off-by: Thomas Mingarelli <Thomas.Mingarelli@hp.com>
Signed-off-by: Wim Van Sebroeck <wim@iguana.be>
Acked-by: Jiri Kosina <jkosina@suse.cz>
---

diff --git a/drivers/watchdog/hpwdt.c b/drivers/watchdog/hpwdt.c
index 70c2c24..af48075 100644
--- a/drivers/watchdog/hpwdt.c
+++ b/drivers/watchdog/hpwdt.c
@@ -443,7 +443,7 @@ static void hpwdt_ping(void)
 static int hpwdt_change_timer(int new_margin)
 {
 	/* Arbitrary, can't find the card's limits */
-	if (new_margin < 30 || new_margin > 600) {
+	if (new_margin < 5 || new_margin > 600) {
 		printk(KERN_WARNING
 			"hpwdt: New value passed in is invalid: %d seconds.\n",
 			new_margin);
